nodes.go

```diff
@@ -620,10 +621,9 @@ func (s *NodesService) putOrOverwrite(in io.Reader, httpVerb, url, name, metadat
 	contentType := writer.FormDataContentType()
 	contentLength := int64(-1)
 
-	buf := make([]byte, 1)
-	n, err := in.Read(buf)
-	isZeroLength := n == 0 && err == io.EOF
-	in = io.MultiReader(bytes.NewReader(buf), in)
+	bufIn := bufio.NewReader(in)
```

Comment:
I'd prefer not to use the bufio as we can't guarantee the read sizes. 

The code should probably say `bytes.NewReader(buf[:n])` shouldn't it?